src/class-big-pit.php

```diff
@@ -79,6 +88,16 @@ public function get( string $key, string $group ): mixed {
 			return null;
 		}
 
+		if ( isset( $this->cache[ $group ] ) && array_key_exists( $key, $this->cache[ $group ] ) ) {
+			$value = $this->cache[ $group ][ $key ];
+
+			if ( is_object( $value ) ) {
+				$value = clone $value;
```

Comment:
I gather that this is an established pattern when accessing objects from an array, but I wouldn't mind a comment here as to why so it's clear to future readers that you didn't intend to assign this clone to a new variable.

Reply:
Good call. I added a comment. It's not about the array, rather about not reusing the object instance.